src/ray/core_worker/reference_count.cc

```diff
@@ -247,6 +247,19 @@ void ReferenceCounter::SetNestedRefInUseRecursive(ReferenceTable::iterator inner
   }
 }
 
+void ReferenceCounter::ReleaseAllLocalReferences() {
+  ReferenceTable all_local_refs;
+  {
+    absl::MutexLock lock(&mutex_);
+    /// A copy of object_id_refs_ to avoid iter invalidation
```

Comment:
This seems like it could race with reference creation. Can you instead hold the lock for the whole method and add a RemoveLocalReferenceInternal that requires the lock to be held?